src/engine.rs

```diff
@@ -307,6 +308,8 @@ where
 
     listeners: Vec<Arc<dyn EventListener>>,
 
+    engine_file: File,
```

Comment:
Put this inside `FilePipeLog`?

Reply:
@tabokie I have try to put it in `FilePipeLog `, but `FilePipeLog ` has marked as `Clone` and File is not clonable and had no idea to solve this problem, so i put it there, do you have any idea about this?

Reply:
You can remove the `#[derive(Clone)]` for `FilePipeLog`, it's not needed now. As a side note, you can make any type `T` cloneable by wrapping it into `Arc<T>`.

Reply:
Got it